src/java/org/apache/cassandra/cql3/functions/UDFContextImpl.java

```diff
@@ -65,11 +68,27 @@ public UDTValue newReturnUDTValue()
         return newUDTValue(returnCodec);
     }
 
+    private KeyspaceMetadata getMetadata()
```

Comment:
It might be worth adding a brief comment about why we need to lazily get the keyspace metadata after instantiation.